pd.lua

```diff
@@ -239,6 +234,11 @@ pd.Class = pd.Prototype:new()
 
 function pd.Class:register(name)
 -- register new class
+  if pd._loadname and nil ~= pd._classes[pd._loadname] then
+    return pd._classes[pd._loadname]
+  elseif nil ~= pd._classes[name] then
```

Comment:
The `  elseif nil ~= pd._classes[name] then` I mean.

Reply:
Hmm no, the way _loadname is defined this probably won't work either. (It's only defined for proper path names, right?)

But I think that the elseif part should only be invoked if _loadname isn't set, or am I missing something there?

Reply:
Anyway, I'm grasping at straws there. I'll have to study your changes tomorrow with a fresh mind.

Reply:
yeah this is the bug I was talking about. But the fact that the classes are initially shared at first is confusing. Because pd won't call 'setup' for the class if it is already created with a pathname.
so if a 'bareword' object is created, it is an instance of the class created w/ a pathname (and this is the way pd does things as well afaict)

Reply:
Yes. I think that to make this work we actually need to always keep track of the proper loadpath of an external, in both the loader hooks and dofile, and in the classes table. That's what I meant when I wrote earlier that a redesign of the class system is in order.

Reply:
I wonder if it would be enough to store the creation name with the object in the objects table..